lib/cc/engine/analyzers/ruby/main.rb

```diff
@@ -24,6 +24,10 @@ class Main < CC::Engine::Analyzers::Base
 
           def process_file(file)
             RubyParser.new.process(File.binread(file), file, TIMEOUT)
+          rescue Timeout::Error
```

Comment:
Semantically, I think a timeout is distinct from a `ParserError`. A parse error (hopefully) indicates invalid syntax (though of course in practice with external parsers it  could also mean a parser bug leading to a crash). A timeout could indicate that the file was just *really* big/complicated, or could also indicate a parser bug.

I think it's valuable to raise/log more specific errors types when available. In a better world, it'd be nice to have more specific sub-classes of `ParserError`, but that's difficult due to the nature of these parsers being external processes. But timeout is a case that we can keep distinct, so I think we should.